### PR TITLE
Writing example for latency

### PR DESCRIPTION
**Pull request by Huy-Ngo**
This is example for `Source.latency`. It prints latency (in nanoseconds) every 10 milliseconds.

### CODE REVIEW

File: examples/palace-latency.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+from argparse import ArgumentParser
+from datetime import timedelta
+from itertools import count, takewhile
+from sys import exit
+from time import sleep
+
+from palace import (
+    Buffer, Context, Device, Source,
+    decode)
```
Comment (McSinyx) [inline examples/palace-latency.py]: This does fit on one line yeah?

File: examples/palace-latency.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+CHUNK_LEN: int = 12000
+QUEUE_SIZE:int = 40
+PERIOD: float = 0.01
+
+
+def play(filename: str, device: str) -> None:
+    """Load and play the file on given device."""
+    with Device(device) as dev, Context(dev) as ctx, Buffer(filename) as buf, Source() as src:
+        decoder = decode(filename)
```
Comment (McSinyx) [inline examples/palace-latency.py]: Use either decoder or buffer, not both.

File: examples/palace-latency.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        decoder.play(CHUNK_LEN, QUEUE_SIZE, src)
+        for i in takewhile(lambda i: src.playing, count()):
+            print(f'Latency at {round(i * PERIOD, 2)}s: ', src.latency, '(ns)')
```
Comment (McSinyx) [inline examples/palace-latency.py]: I think you can use `Source.offset[_seconds]` instead of manually computing it; also do we really need ns precision?  Milliseconds seems fine to me.

Edit:  Because of this,  the loop can be simplified as `while src.playing`.  Please visit other examples where the same could be apply and fix them in a separate commit.

File: examples/palace-latency.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+
+if __name__ == '__main__':
+    parser = ArgumentParser()
+    parser.add_argument('filename', help='audio files')
+    parser.add_argument('-d', '--device', default='', help='device name')
+    args = parser.parse_args()
+    filename = args.filename
+    device = args.device
+    play(filename, device)
```
Comment (McSinyx) [inline examples/palace-latency.py]: WTF is this :smiley:  BTW I think it's not difficult to support multiple audio files playback.

File: examples/palace-latency.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# Example for latency checking
+# Copyright (C) 2020 Ngô Ngọc Đức Huy
```
Comment (McSinyx) [inline examples/palace-latency.py]: Extreme nit: 2 spaces between year and name.

**Review by McSinyx — CHANGES_REQUESTED** [review]
Please see `flake8` output and edit accordingly.

File: examples/palace-latency.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        for filename in files:
+            try:
+                decoder = decode(filename)
+            except RuntimeError:
+                stderr.write(f'Failed to open file: {filename}\n')
+            decoder.play(CHUNK_LEN, QUEUE_SIZE, src)
+            print('Playing: ', filename)
+            for i in takewhile(lambda i: src.playing, count()):
+                print('Offset:', round(src.offset_seconds),
+                      '-',
```
Comment (McSinyx) [inline examples/palace-latency.py]: Ugh!

LGTM otherwise.  Also rmb to fix the `for i in takewhile(lambda i: src.playing, count())` in examples that do not use `i`, preferably in a separate commit.

File: examples/palace-latency.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                      'Latency:', src.latency//10**6, 'ms',
+                      end='\r', flush=True)
+                sleep(PERIOD)
+                ctx.update()
+        print()
```
Comment (McSinyx) [inline examples/palace-latency.py]: Oops this is in the wrong scope.

File: examples/palace-latency.py (hunk 6 above)
Comment (Huy-Ngo) [reply examples/palace-latency.py]: Wait, which one doesn't use `i`?

File: examples/palace-latency.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+from argparse import ArgumentParser
+from itertools import count, takewhile
+from sys import stderr
+from time import sleep
+from typing import Iterable
+
+from palace import Context, Device, Source, decode
+
+CHUNK_LEN: int = 12000
+QUEUE_SIZE: int = 40
```
Comment (McSinyx) [inline examples/palace-latency.py]: I'm poor, please lower this.  Otherwise you can squash and merge now.

**Comment by McSinyx** [comment]
Now please fix the loop things.

**Comment by Huy-Ngo** [comment]
Which loop things? All examples use `for i in takewhile(lambda i: src.playing, count())`, except for `palace-hrtf` which use `count(PERIOD)` instead.

**Comment by McSinyx** [comment]
Any of those loop that does not use `i` within could be replaced by `while src.playing`.
